code/script.js

```diff
@@ -1,9 +1,86 @@
-// A function that adds and remove the class "active" on the section you click on.
-// We haven't really talked about ´this´ yet, but we will... ;)
-// console.log(this) to see how it works
-function toggle() {
-  this.classList.toggle("active")
+
+const question = document.querySelectorAll(".question")  /* querySelectorAll returns a Nodelist object collection 
+- in other words all element with exact class name/names*/
+let i; /* declaring i */
+
+for (i = 0; i < question.length; i++) {  /* here is for loop starts by assigning an initial expression (value) of 0 to the "i" variable , 
+  "i < question.length" is a conditional expresion that specifies that "i" cant go over the length of "question" (array), 
+  and i++ (increment operator ++) means that each time loop is activated 1 will be added to the "i" variable */
+  
+  question[i].addEventListener("click", function() { /*here an "Event listener" listens for a user's "click" and triggers the function
+  for every element in the "question" array  */
+
+  // This nested for loop is aimed to check if the current "question" is the same as user had clicked on or not.
+  // if it is the same one, then class "active" is toggled on the "question". if it is not the one that was selected by user 
+  // then class active is removed from the question.
+   for (let j=0; j<question.length; j++) { 
+   if (question[j] === this) {
+    this.classList.toggle("active");
+   } else {question[j].classList.remove("active")}  
+   }
+  });
 }
```

Comment:
I really like this function. I will definitely try this for my future projects, thank you for the well written description.

Reply:
Happy to here )